BedrockServer.cpp

```diff
@@ -713,385 +705,399 @@ void BedrockServer::worker(int threadId)
+        if (command->repeek || !command->httpsRequests.size()) {
+            peekResult = core.peekCommand(command, isBlocking);
+            calledPeek = true;
+        }
+
+        if (!calledPeek || peekResult == BedrockCore::RESULT::SHOULD_PROCESS) {
+            // We've just unsuccessfully peeked a command, which means we're in a state where we might want to
+            // write it. We'll flag that here, to keep the node from falling out of LEADING/STANDINGDOWN
+            // until we're finished with this command.
+            if (command->httpsRequests.size()) {
+                // This *should* be impossible, but previous bugs have existed where it's feasible that we call
+                // `peekCommand` while leading, and by the time we're done, we're FOLLOWING, so we check just
+                // in case we ever introduce another similar bug.
+                if (state != SQLiteNode::LEADING && state != SQLiteNode::STANDINGDOWN) {
+                    SALERT("Not leading or standing down (" << SQLiteNode::stateName(state)
+                           << ") but have outstanding HTTPS command: " << command->request.methodLine
+                           << ", returning 500.");
+                    command->response.methodLine = "500 STANDDOWN TIMEOUT";
+                    _reply(command);
+                    core.rollback();
+                    break;
+                }
 
-                            if (!command->areHttpsRequestsComplete()) {
-                                // If it has outstanding HTTPS requests, we'll wait for them.
-                                waitForHTTPS(move(command));
-                            } else if (command->repeek) {
-                                // Otherwise, it needs to be re-peeked, but had no outstanding requests, so it goes
-                                // back in the main queue.
-                                commandQueue.push(move(command));
-                            }
+                // If the command isn't complete, we'll re-queue it.
+                if (command->repeek || !command->areHttpsRequestsComplete()) {
+                    // Roll back the existing transaction, but only if we are inside an transaction
+                    if (calledPeek) {
+                        core.rollback();
+                    }
 
-                            // Move on to the next command until this one finishes.
-                            break;
-                        }
+                    if (!command->areHttpsRequestsComplete()) {
+                        // If it has outstanding HTTPS requests, we'll wait for them.
+                        waitForHTTPS(move(command));
+                    } else if (command->repeek) {
+                        // Otherwise, it needs to be re-peeked, but had no outstanding requests, so it goes
+                        // back in the main queue.
+                        _commandQueue.push(move(command));
```

Comment:
If I'm reading this correctly, this is a change in behavior - previously, `commandQueue` was set to either `_commandQueue` or `_blockingCommandQueue`, and now we're only using `_commandQueue`. Is this change intentional?

Reply:
Yes. This doesn't actually matter. The preexisting comment implied we thought this is what we had been doing the whole time.

Nothing should be getting `repeek`ed in the blocking queue regardless. We should only ever `repeek` anything before running `process`, and we only every move anything to the blocking queue after a commit conflict, which is inherently after `process` has happened.

Even if we did move a command back from the blocking to the main queue, it would just give it a few more opportunities to conflict before making it back to the blocking queue.

Also, the blocking queue is going to go away anyway as soon as this change is done.